component/vshn_postgres.jsonnet

```diff
@@ -0,0 +1,231 @@
+
+local defaultDB = 'postgres';
+local defaultUser = 'postgres';
+local defaultPort = '5432';
+
+
+local connectionSecretKeys = [
+  'POSTGRESQL_URL',
+  'POSTGRESQL_DB',
+  'POSTGRESQL_HOST',
+  'POSTGRESQL_PORT',
+  'POSTGRESQL_USER',
+  'POSTGRESQL_PASSWORD',
+  'ca.crt',
+];
+
+local xrd = xrds.XRDFromCRD(
+  'xvshnpostgresqls.vshn.appcat.vshn.io',
+  xrds.LoadCRD('vshn.appcat.vshn.io_vshnpostgresqls.yaml'),
+  defaultComposition='vshnpostgres.vshn.appcat.vshn.io',
+  connectionSecretKeys=connectionSecretKeys,
+);
+
+
+local composition =
+  local namespace = comp.KubeObject('v1', 'Namespace') +
+                    {
+                      spec+: {
+                        forProvider+: {
+                          manifest+: {
+                            metadata: {
+                              name: '',
+                            },
+                          },
+                        },
+                      },
+                    };
+
+  local sgInstanceProfile = comp.KubeObject('stackgres.io/v1', 'SGInstanceProfile') +
+                            {
+                              spec+: {
+                                forProvider+: {
+                                  manifest+: {
+                                    metadata: {
+                                      name: '',
+                                      namespace: '',
+                                    },
+                                    spec: {
+                                      cpu: '',
+                                      memory: '',
+                                    },
+                                  },
+                                },
+                              },
+                            };
+
+  local sgPostgresConfig = comp.KubeObject('stackgres.io/v1', 'SGPostgresConfig') +
+                           {
+                             spec+: {
+                               forProvider+: {
+                                 manifest+: {
+                                   metadata: {
+                                     name: '',
+                                     namespace: '',
+                                   },
+                                   spec: {
+                                     postgresVersion: '',
+                                     'postgresql.conf': {},
+                                   },
+                                 },
+                               },
+                             },
+                           };
+
+  local sgCluster = comp.KubeObject('stackgres.io/v1', 'SGCluster') +
+                    {
+                      spec+: {
+                        forProvider+: {
+                          manifest+: {
+                            metadata: {
+                              name: '',
+                              namespace: '',
+                            },
+                            spec: {
+                              instances: 1,
```

Comment:
is there a reason why the value of instances is 1? shouldn't it be patched?

Reply:
We decided to hard code it to 1 for now.